patches/custom-extensions-gallery.patch

```diff
@@ -0,0 +1,28 @@
+diff --git a/src/vs/platform/product/common/product.ts b/src/vs/platform/product/common/product.ts
+index 2bea85740c..d0e9da036c 100644
+--- a/src/vs/platform/product/common/product.ts
++++ b/src/vs/platform/product/common/product.ts
+@@ -57,6 +57,23 @@ else {
+ 		});
+ 	}
+ 
++	// Set user-defined extension gallery
++	if (
++		env['VSCODIUM_EXTENSIONS_GALLERY_SERVICE_URL']
++		|| env['VSCODIUM_EXTENSIONS_GALLERY_ITEM_URL']
++		|| env['VSCODIUM_EXTENSIONS_GALLERY_CONTROL_URL']
++		|| env['VSCODIUM_EXTENSIONS_GALLERY_RECOMMENDATIONS_URL']
++	) {
++		Object.assign(product, {
++			extensionsGallery: {
```

Comment:
Using `||` here doesn't seem safe, since some of the vars are mandatory. Maybe check if `_SERVICE_URL` and `_ITEM_URL` are both set, and proceed with overriding `extensionsGallery` only if they are?